packages/mdl-drawer/temporary/foundation.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings, numbers} from './constants';
+
+export default class MDLTemporaryDrawerFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get numbers() {
+    return numbers;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hasClass: (/* className: string */) => {},
+      hasNecessaryDOM: () => /* boolean */ {},
+      registerComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerTransitionEndHandler: (/* handler: EventListener */) => {},
+      deregisterTransitionEndHandler: (/* handler: EventListener */) => {},
+      updateDrawerPosition: (/* value: number | null */) => {},
+      updateComponentBackground: (/* value: string */) => {},
+      getFocusableElements: () => /* array<Element> */ {},
```

Comment:
supernit: Typescript / Flow / ES4 [represent arrays](https://www.typescriptlang.org/docs/handbook/basic-types.html) as `T[]`, e.g. `Element[]`. Should we do the same?